### PR TITLE
Add color picker for syntax classification

### PR DESCRIPTION
**Pull request by ryzngard**
Syntax classification and the settings for it aren't really easy to find in Tools > Options > Fonts and Colors. This introduces an easier way to set colors by following the syntax tree. If a TextSpan is classified (using our public API for classification) then we use it to get the VS color settings for that classification and display to the user, allowing it to be changed. 

Most of the code changes are adding a color picker since I couldn't find one in VS or WPF readily. Heavily motivated from https://github.com/dsafa/wpf-color-picker 

Fonts and colors settings ported with help from @JoeRobich 

# Blue Theme Example

classification_themed_blue_mode

# Dark Theme Example

classification_themed_dark_mode

### CODE REVIEW

File: src/VisualStudio.Roslyn.SDK/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Control/FontsAndColorsHelper.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+                if (fontAndColorUtilities.GetColorType(foregroundColorRef, out var foregroundColorType) != VSConstants.S_OK)
+                {
+                    // Without being able to check color type, we cannot make a determination.
+                    return null;
+                }
+
+                // Since the color type isn't default then it has been customized, we will
+                // perform an additional check for RGB colors to see if the customized color
+                // matches the color scheme color.
```
Comment (JoeRobich) [inline src/VisualStudio.Roslyn.SDK/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Control/FontsAndColorsHelper.cs:58]: This comment can be removed.

File: src/VisualStudio.Roslyn.SDK/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Control/FontsAndColorsHelper.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            var fontsAndColorStorage = ServiceProvider.GlobalProvider.GetService<SVsFontAndColorStorage, IVsFontAndColorStorage>();
+
+            if (fontsAndColorStorage is null)
+            {
+                return;
+            }
+
+            // Open Text Editor category for readonly access.
```
Comment (JoeRobich) [inline src/VisualStudio.Roslyn.SDK/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Control/FontsAndColorsHelper.cs:89]: This comment needs updating.

**Review by JoeRobich — APPROVED** [review]
lgtm. just some stale comments

File: src/VisualStudio.Roslyn.SDK/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Control/Roslyn.SyntaxVisualizer.Control.csproj [hunk 3]
```diff
@@ -15,6 +15,7 @@
     <PackageReference Include="Microsoft.CodeAnalysis.Common" Version="$(MicrosoftCodeAnalysisCommonVersionVS2019)" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="$(MicrosoftCodeAnalysisCSharpVersionVS2019)" />
     <PackageReference Include="Microsoft.CodeAnalysis.VisualBasic" Version="$(MicrosoftCodeAnalysisVisualBasicVersionVS2019)" />
+    <PackageReference Include="Microsoft.CodeAnalysis.Workspaces.Common" Version="3.3.1" />
```
Comment (sharwell) [inline src/VisualStudio.Roslyn.SDK/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Control/Roslyn.SyntaxVisualizer.Control.csproj:18]:
```suggestion
    <PackageReference Include="Microsoft.CodeAnalysis.Workspaces.Common" Version="$(MicrosoftCodeAnalysisWorkspacesCommonVersionVS2019)" />
```

**Review by sharwell — CHANGES_REQUESTED** [review]
Marking Request Changes so we have an opportunity to verify two things:

1. Does the new dialog need to itself be dark-themed when the dark theme is enabled?
2. What is the accessibility state and story?

**Comment by ryzngard** [comment]
@sharwell  good call on themeing. I updated the UI to be themed and updated the gifs in the summary. Accessibility I'm following up on with the accessibility champ

**Comment by ryzngard** [comment]
Merging. @sharwell  this has undergone initial accessibility testing and UX review. There may be additions to the UI in a further review, but this has passed.  Will do a manual test pass with team after checkin per Abhitej's request

**Pull request merged**